service/history/historyCache.go

```diff
@@ -72,38 +72,65 @@ func newHistoryCache(shard ShardContext) *historyCache {
 	}
 }
 
-func (c *historyCache) getOrCreateWorkflowExecution(domainID string,
-	execution workflow.WorkflowExecution) (workflowExecutionContext, releaseWorkflowExecutionFunc, error) {
-	return c.getOrCreateWorkflowExecutionWithTimeout(context.Background(), domainID, execution)
-}
+func (c *historyCache) getOrCreatCurrentWorkflowExecution(
+	ctx context.Context,
+	domainID string,
+	workflowID string,
+) (workflowExecutionContext, releaseWorkflowExecutionFunc, error) {
 
-func (c *historyCache) validateWorkflowExecutionInfo(domainID string, execution *workflow.WorkflowExecution) error {
-	if execution.GetWorkflowId() == "" {
-		return &workflow.BadRequestError{Message: "Can't load workflow execution.  WorkflowId not set."}
+	// using empty run ID as current workflow run ID
+	runID := ""
+	execution := workflow.WorkflowExecution{
+		WorkflowId: common.StringPtr(workflowID),
+		RunId:      common.StringPtr(runID),
 	}
 
-	// RunID is not provided, lets try to retrieve the RunID for current active execution
-	if execution.GetRunId() == "" {
-		response, err := c.getCurrentExecutionWithRetry(&persistence.GetCurrentExecutionRequest{
-			DomainID:   domainID,
-			WorkflowID: execution.GetWorkflowId(),
-		})
+	// Test hook for disabling the cache
+	if c.disabled {
+		return newWorkflowExecutionContext(
+			domainID,
+			execution,
+			c.shard,
+			c.executionManager,
+			c.logger,
+		), func(error) {}, nil
+	}
 
+	key := definition.NewWorkflowIdentifier(domainID, workflowID, runID)
+	workflowCtx, cacheHit := c.Get(key).(workflowExecutionContext)
+	if !cacheHit {
+		c.metricsClient.IncCounter(metrics.HistoryCacheGetOrCreateScope, metrics.CacheMissCounter)
```

Comment:
scope name should be different given that this is a different method

Reply:
good catch